stdlib/public/Differentiation/ArrayDifferentiation.swift

```diff
@@ -74,11 +74,12 @@ where Element: Differentiable {
 
   public mutating func move(by offset: TangentVector) {
     precondition(
-      base.count == offset.base.count, """
+      offset.base.isEmpty || base.count == offset.base.count,
+        """
         Count mismatch: \(base.count) ('self') and \(offset.base.count) \
         ('direction')
         """)
-    for i in base.indices {
+    for i in offset.base.indices {
       base[i].move(by: offset.base[i])
     }
```

Comment:
I'd suggest adding an early return before `precondition` and reverting the changes to the precondition.
```swift
if offset.base.isEmpty {
  return
}
```

Reply:
Sure thing.